sketch/sketch_base.hpp

```diff
@@ -8,8 +8,13 @@ namespace ts {
 template <typename sketch_type_, bool kmer_input_>
```

Comment:
not sure what's the point of using the `xxx_` name and then alias `xxx`,why not just use one name?

Reply:
Yeah it's annoying: You want to do `SketchBase::sketch_type`, but you can't access template types that way. And also they can't be the same, because `using X = X;` will complain...